services/meta/client.go

```diff
@@ -834,12 +834,17 @@ func (c *Client) JoinMetaServer(httpAddr, tcpAddr string) error {
 			currentServer++
 			continue
 		}
-		resp.Body.Close()
 
 		// Successfully joined
 		if resp.StatusCode == http.StatusOK {
-			return nil
+			defer resp.Body.Close()
```

Comment:
That won't close the body in the case there is no error.  I'll convert encoder to one line though.